File: Level3/Materal.BusinessFlow/Demo/BusinessFlowWebDemo/src/models/DataModelField/DataModelField.ts
import { BaseDomain } from "../BaseDomain";
import { DataTypeEnum } from "./DataTypeEnum";

export class DataModelField extends BaseDomain {
    /**
     * 名称
     */
    public Name: string = "";
    /**
     * 数据模型ID
     */
    public DataModelID: string = "";
    /**
     * 数据类型
     */
    public DataType: DataTypeEnum = DataTypeEnum.String;
    /**
     * 数据类型
     */
    public DataTypeText: String = "";
    /**
     * 数据
     */
    public Data?: string;
    /**
     * 描述
     */
    public Description?: string;
}

File: Level3/Materal.BusinessFlow/Src/Materal.BusinessFlow.Abstractions/Domain/DataModelField.cs
﻿using System.ComponentModel.DataAnnotations;

namespace Materal.BusinessFlow.Abstractions.Domain
{
    /// <summary>
    /// 数据模型字段
    /// </summary>
    public class DataModelField : BaseDomain
    {
        /// <summary>
        /// 名称
        /// </summary>
        [Required, StringLength(40)]
        public string Name { get; set; } = string.Empty;
        /// <summary>
        /// 数据模型唯一标识
        /// </summary>
        [Required]
        public Guid DataModelID { get; set; }
        /// <summary>
        /// 数据类型
        /// </summary>
        [Required]
        public DataTypeEnum DataType { get; set; } = DataTypeEnum.String;
        /// <summary>
        /// 数据
        /// </summary>
        public string? Data { get; set; }
        /// <summary>
        /// 描述
        /// </summary>
        public string? Description { get; set; }
    }
}
